src/main/java/betterquesting/client/gui2/GuiQuestLines.java

```diff
@@ -354,32 +361,63 @@ public void initPanel() {
-                            }
-                            if (integer <= 1) {
-                                claimAll();
-                            }
-                        },
-                        QuestTranslation.translate("gui.yes"),
-                        QuestTranslation.translate("betterquesting.gui.yes_always"),
-                        QuestTranslation.translate("gui.no")));
+            if (BQ_Settings.claimAllConfirmation || isShiftKeyDown()) {
+                Pair<List<UUID>, Integer> data = getAllPossibleClaims();
+                List<UUID> claimIdList = data != null ? data.getLeft() : null;
+                int numChoiceQuests = data != null ? data.getRight() : 0;
+
+                PopChoiceExt popup = new PopChoiceExt(
+                    QuestTranslation.translate("betterquesting.gui.claim_all_warning") + "\n\n"
+                        + QuestTranslation.translate("betterquesting.gui.claim_all_confirm")
+                        + "\n\n"
+                        + QuestTranslation.translate(
+                            "betterquesting.gui.claim_all_quest_count",
+                            claimIdList != null ? claimIdList.size() : 0)
+                        + "\n"
+                        + QuestTranslation
+                            .translate("betterquesting.gui.claim_all_choice_quest_count", numChoiceQuests),
+                    PresetIcon.ICON_CHEST_ALL.getTexture());
+
+                popup.addOption(
+                    QuestTranslation.translate("gui.yes"),
+                    btn -> claimAll(claimIdList, BQ_Settings.claimAllRandomChoice),
+                    true);
+                popup.addOption(QuestTranslation.translate("betterquesting.gui.yes_always"), btn -> {
+                    ConfigHandler.config.get(Configuration.CATEGORY_GENERAL, "Claim all requires confirmation", true)
+                        .set(false);
+                    ConfigHandler.config.save();
+                    ConfigHandler.initConfigs();
+                    claimAll(claimIdList, BQ_Settings.claimAllRandomChoice);
+                }, true);
+                popup.addOption(QuestTranslation.translate("gui.no"), null, true);
+
+                String startingChoice = BQ_Settings.claimAllRandomChoice ? "betterquesting.gui.force_choice_yes"
+                    : "betterquesting.gui.force_choice_no";
+                popup.addOption(QuestTranslation.translate(startingChoice), btn -> {
+                    // Always save the result of the checkbox, this way it is "remembered" for next time
+                    Property prop = ConfigHandler.config
+                        .get(Configuration.CATEGORY_GENERAL, "Claim all random select choice rewards", false);
+                    prop.set(!prop.getBoolean());
+                    ConfigHandler.config.save();
+                    ConfigHandler.initConfigs();
+
+                    String resultChoice = BQ_Settings.claimAllRandomChoice ? "betterquesting.gui.force_choice_yes"
```

Comment:
We have 2 instances of this ternary operation. Extract to a getForceChoiceString method?